azurerm/internal/services/apimanagement/resource_arm_api_management.go

```diff
@@ -907,9 +919,20 @@ func expandAzureRmApiManagementIdentity(d *schema.ResourceData) *apimanagement.S
 
 	v := vs[0].(map[string]interface{})
 	identityType := v["type"].(string)
-	return &apimanagement.ServiceIdentity{
+
+	identityIds := make(map[string]*apimanagement.UserIdentityProperties)
+	for _, id := range v["identity_ids"].([]interface{}) {
+		identityIds[id.(string)] = &apimanagement.UserIdentityProperties{}
+	}
+	managedServiceIdentity := apimanagement.ServiceIdentity{
 		Type: apimanagement.ApimIdentityType(identityType),
 	}
+
+	if managedServiceIdentity.Type == apimanagement.UserAssigned || managedServiceIdentity.Type == apimanagement.SystemAssignedUserAssigned {
```

Comment:
Could we throw an error when a user is attempting to set user identity ids while the type is set to `SystemAssigned`?
You can take [this](https://github.com/terraform-providers/terraform-provider-azurerm/blob/b2bc8ea46dd0b2d93c6ec8c0920245ef17689c6e/azurerm/internal/services/compute/virtual_machine.go#L104) as an example.

Reply:
Yes, I've made that change now in 3a2bf09.